### DISCUSSION TITLE
Ativar i18n

### DISCUSSION THREAD

**Discussion by shonorio** [institutoazmina/penhas-app#130] [Ideas]
Atualmente, todos os textos de interface com o usuário estão "hardcoded" no aplicativo, uma abordagem que não é considerada uma prática recomendada em arquitetura de software. Essa escolha foi feita na fase inicial do projeto com o objetivo de acelerar o desenvolvimento.

Para melhorar a flexibilidade e a manutenibilidade do código, recomendo a implementação da internacionalização (i18n) no aplicativo. O plano é executar essa migração de forma gradual para minimizar o impacto na base de código existente e no fluxo de trabalho da equipe.

Plano de Ação Proposto:

1. Iniciar um subprojeto específico focado na migração para i18n.
2. Realizar um mapeamento completo das telas, widgets e mensagens atualmente em "hardcode".
3. Criar tarefas individuais para a transição de cada elemento identificado para o sistema i18n.
4. Monitorar e avaliar periodicamente o progresso da migração, ajustando o plano conforme necessário.

Com essa estrutura organizada, poderemos não apenas rastrear a evolução da atividade, mas também preparar o aplicativo para suportar múltiplos idiomas e configurações regionais no futuro.

**Comment by pedrox-hs** [comment]
Eu apoio! Acredito que seria ótimo fazer isso, ficariamais fácil de rastrear as strings e teriamos muitas outras vantagens.

@diraol e @hsvab, conseguiriamos colocar isso em pauta nas próximas reuniões?
